src/wrover_status_light/ServiceIntro.md

```diff
@@ -0,0 +1,160 @@
+```
+ExmaplePackage::ExampleService exampleService;
+ros::NodeHandle nh;
+ros::ServiceServer service = nh.advertiseService("example_service", &ExampleServiceServer::callblack, &exampleService);
+```
+The code sets up the service server using a `NodeHandle` instance. NodeHandles provide functions for creating, managing, and interacting using ROS nodes. Here the `NodeHandle` creates a new server with `example_service` as the server name, `ExmapleServiceServer::callback` as the handler function for service calls, and `exmapleService` defines teh service type. 
+
+## Writing a Client Node
+### Python
+Services can be called by 
+```
+rospy.wait_for_service('example_service')
+example_service = rospy.ServiceProxy('example_service', ExamplePackage.srv.ExampleService)
+response = example_service(ExamplePackage.srv.ExampleServiceRequest(..., ...))
+```
+To break the code down,
+```
+rospy.wait_for_service('exmaple_service')
+```
+This is a blocking operation that waits until a service with the name `example_service` is created. 
+```
+example_service = rospy.ServiceProxy('example_service', ExamplePackage.srv.ExampleService)
+```
+Creates a new callable proxy object for the service named `example_service` and defined by `ExmaplePackage.srv.ExampleService`.
+```
+response = example_service(ExamplePackage.srv.ExmapleServiceRequest(..., ...))
+```
+Calls the service using an instance of the request message class and saves the response in a local variable. Similar to before, the service can also be called without explicitly making a new instance with
+```
+response = exmaple_service(..., ...)
+```
+
+### C++
+Services can be called by
+```
+ros::NodeHandle nh;
+ros::ServiceClient client = nh.serviceClient<Example_Package::Example_Service>("example_service");
+
+Example_Package::Example_Service srv;
+srv.request.input1 = ...
+srv.request.input2 = ...
+if (client.call(srv)) {
+    ...
+} else {
+    ...
+}
+```
+To break the code down
+```
+ros::NodeHandle nd;
+ros::ServiceClient client = nh.serviceClient<Example_Package::Example_Service>("example_service");
+```
+The code first creates a new NodeHandle instance. Here the NodeHandle instance creates a new ServiceClient using the type `Example_Package::Example_Service` and service name `example_service`.
+```
+Example_Package::Example_Service srv;
+srv.request.input1 = ...
+srv.request.input2 = ...
+```
+The code then creates a new `Example_Service` instance and populates the request variables.
+```
+if (client.call(srv)) {
+    ...
+} else {
+    ...
+}
+```
+Lastly, the code calls the service using the `call()` function. If the service call is successful, then the response value can be accessed using `srv.response`. If it is unsuccessful, then the code will go into the else block.
+
+## Notes
+ROS's wiki page on services: http://wiki.ros.org/Services
```

Comment:
Maybe point to the provided tutorials as well for more example code.

Reply:
Addressed in commit [cc1f8e6](https://github.com/WisconsinRobotics/WRoverPlayground/pull/9/commits/cc1f8e6a27e5b1774be76c98630bef821db9c083). Let me know if you meant other tutorials.